lib/puppet/util/windows/eventlog.rb

```diff
@@ -0,0 +1,114 @@
+require 'puppet/util/windows'
```

Comment:
Might have to dig into to it to understand what it's capable of easily loading without a lot of hackery.  Obviously the current daemon is loading gems (i.e. `win32-eventlog` / `win32-service`) using `require` so it has access to the same Ruby env that Puppet has.

I think that means is could theoretically `require 'puppet/util/windows/api_types.rb'` and share the same files, but not sure if that's confusing to maintain.  I think we don't do a `require 'puppet'` because it drags along a bunch of startup initialization / settings parsing that clearly isn't needed for a simple service runner.  But it's possible we could load just a few files we need (as long as they're isolated, and we can make sure through tests that the service doesn't load more code than necessary).